### PR TITLE
cmd/govim: add tests for gopls features where files are initially empty

### PR DESCRIPTION
**Pull request by myitcv**
We have a number of tests that check gopls behaviour where files do not
exist on disk, i.e. they are just buffers in memory. But we are missing
tests to verify that if .go files on disk are empty and then "filled in"
by changes in the editor, that things work as expected.

Add some tests of that sort.

### CODE REVIEW

File: cmd/govim/testdata/scenario_default/quickfix_empty_files.txt [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+# The scenario involves creating a package p with a simple function DoIt. p is
+# imported by a main package. p also has a test file that exercises DoIt, and
+# an external test file that does the same.
+#
+# Initially all call sites for DoIt incorrectly pass an integer argument,
+# meaning we should have error diagnostics for all call sites. Then we correct
+# the definition of DoIt to take an integer argument at which point all
+# diagnostics should disappear.
+
+# skip 'Pending fix for golang.org/issues/36795'
```
Comment (leitzler) [inline cmd/govim/testdata/scenario_default/quickfix_empty_files.txt]: I guess this is commented out to provide details to Rebecca & co(?). If so, don't forget to uncomment before merge (or remove if it's fixed).

Comment (myitcv) [reply cmd/govim/testdata/scenario_default/quickfix_empty_files.txt]: Done

**Comment by myitcv** [comment]
Mac builds seem to be taking an inordinate amount of time. Merging.
